Dockerfile

```diff
@@ -1,5 +1,5 @@
 # Use the official lightweight Ruby image
-FROM ruby:slim
+FROM ruby:3.4.2-slim
```

Comment:
:thought_balloon: Good catch. Should this be `3.4` instead of `3.4.2` because pinning this at the patch version shouldn't matter all that much especially since the Ruby core team is publishing matchings on a ~2 month schedule so we'll not want to update this each time a new patch rolls out.

Reply:
Yeah, I don't like that game either, but with the .ruby-version file added, the docker build will break unless it's also updated. I tried this by changing the .ruby-version file to 3.4.1 and trying the docker build, but bundler threw an error instead.

If we could specify a looser ruby version in the Gemfile instead, we could use 3.4.

Reply:
Ah, right....because the image is tied to the specific Ruby version. Keep forgetting about that. OK, let's roll with what you have for now.